app/src/main/java/com/example/apkproject/MainActivity.kt

```diff
@@ -48,14 +75,60 @@ class MainActivity : AppCompatActivity() {
 
                 override fun messageArrived(topic: String?, message: MqttMessage?) {
                     println("$topic: ${message?.payload?.toString(Charsets.UTF_8)}")
-                }
+                    val msg = message?.payload?.toString(Charsets.UTF_8)
+                   runOnUiThread{
+                        when(msg){
+                            "next" -> showNextImage()
```

Comment:
Hardcoded string, you should to be create `Constants` and you need to create value for "next"